dask/dataframe/tests/test_dataframe.py

```diff
@@ -382,25 +324,108 @@ def test_arithmetic_frame():
                                    index=[5, 6, 8]),
             ('y', 2): pd.DataFrame({'a': [1, 4, 10], 'b': [1, 0, 5]},
                                    index=[9, 9, 9])}
-
     d4 = dd.DataFrame(dsk4, 'y', ['a', 'b'], [0, 4, 9, 9])
     full4 = d4.compute()
 
+    # Arithmetics
     cases = [(d, d, full, full),
+             (d, d.repartition([0, 1, 3, 6, 9]), full, full),
              (d2, d3, full2, full3),
-             (d, d4, full, full4)]
+             (d2.repartition([0, 3, 6, 7]), d3.repartition([0, 7]), full2, full3),
+             (d2.repartition([0, 7]), d3.repartition([0, 2, 4, 5, 7]), full2, full3),
+             (d, d4, full, full4),
+             (d, d4.repartition([0, 9]), full, full4),
+             (d.repartition([0, 3, 9]), d4.repartition([0, 5, 9]), full, full4)]
+
     for (l, r, el, er) in cases:
+        check_series_arithmetics(l.a, r.b, el.a, er.b)
+        check_frame_arithmetics(l, r, el, er)
+
+    # different index, pandas raises ValueError in comparison ops
+
+    dsk5 = pd.DataFrame({'a': [3, 2, 1, 5, 2, 8, 1, 4, 10],
+                         'b': [7, 8, 9, 4, 2, 3, 1, 0, 5]},
+                        index=[0, 1, 3, 5, 6, 8, 9, 9, 9])
+    d5 = dd.from_pandas(dsk5, 2)
+    full5 = d5.compute()
```

Comment:
The names here seem odd.  `dsk` refers to a pandas dataframe.  

Why have both `dsk5` and `full5`?

When I have both a pandas and dask dataframe I tend to use the pronounds `df` and `ddf`.  How about `df5` and `ddf5`?